### ISSUE TITLE
`viennacl::matrix_range` methods for solvers (e.g. qr_method)?

### ISSUE THREAD

**Issue by cdeterman** [viennacl/viennacl-dev#175]
Currently it appears to be impossible to use solvers such `qr_method` with `viennacl::matrix_range` objects.  

>error: no matching function for call to ‘qr_method(viennacl::matrix_range<viennacl::matrix<int, viennacl::row_major, 1u> >&, viennacl::matrix_range<viennacl::matrix<int, viennacl::row_major, 1u> >&, std::vector<int>&, std::vector<int>&, bool&)’
     viennacl::linalg::detail::qr_method(vcl_A, vcl_Q, D, E, symmetric);

I can imagine this being useful if a user has matrices with additional columns (such as an intercept column perhaps) that they would wish to exclude and apply a solver.

If you agree, I am not sure how you would want the backend to be modified.  Would you want to have additional methods explicitly defined for `viennacl::matrix_range` objects or would it be possible to generalize to the base `viennacl::matrix_base` class?

**Comment by karlrupp** [comment]
yes, I confirm the issue. The implementation was written at a time when ViennaCL only supported full matrices, no submatrices. It is one of the reasons why our implementation of the QR method is listed as unstable.

Since all unstable parts should become stable by the 1.8.0 release, I scheduled this issue for the 1.8.0 release.

**Comment by cdeterman** [comment]
Excellent, I know 1.7.0 was released relatively recently but do you have an estimated timeline on 1.8.0?  Not intending to rush, just curious for my own development timeline.

**Comment by karlrupp** [comment]
1.8.0 should be out ASAP ;-)

To be more precise: 1.7.1 will be released in the next few days. 1.8.0 will most likely follow in the second half of February.
